File: backend/schemas/filter.py
from typing import Optional
from sqlmodel import SQLModel
from .categories import States

class Filter(SQLModel):
    id: Optional[str] = None
    status: Optional[str] = None
    created_after: Optional[float] = None
    updated_after: Optional[float] = None
    created_before: Optional[float] = None
    updated_before: Optional[float] = None
    categories: Optional[str] = None
    state: Optional[States] = None
    root: Optional[str] = None
    user_id: Optional[str] = None

File: frontend/src/filters/filter.ts
export interface IFilter {
    id?: string;
    status?: string;
    created_after?: number;
    updated_after?: number;
    created_before?: number;
    updated_before?: number;
    categories?: string;
    state?: EStates;
    root?: string;
}

export enum EStates {
    NEW = 1,
    CLARIFICATION = 2,
    RECOMMENDATION = 3,
    WIP = 4,
    READY = 5,
    REJECTED = 6
}